lms/djangoapps/grades/models.py

```diff
@@ -373,95 +361,50 @@ def update_or_create_grade(cls, **params):
         """
         Wrapper for objects.update_or_create.
         """
-        cls._prepare_params_and_visible_blocks(params)
+        cls._prepare_params(params)
+        VisibleBlocks.cached_get_or_create(params['visible_blocks'])
+        cls._prepare_params_visible_blocks_id(params)
+        cls._prepare_params_override(params)
 
         first_attempted = params.pop('first_attempted')
         user_id = params.pop('user_id')
         usage_key = params.pop('usage_key')
 
-        # apply grade override if one exists before saving model
-        try:
-            override = PersistentSubsectionGradeOverride.objects.get(
-                grade__user_id=user_id,
-                grade__course_id=usage_key.course_key,
-                grade__usage_key=usage_key,
-            )
-            if override.earned_all_override is not None:
-                params['earned_all'] = override.earned_all_override
-            if override.possible_all_override is not None:
-                params['possible_all'] = override.possible_all_override
-            if override.earned_graded_override is not None:
-                params['earned_graded'] = override.earned_graded_override
-            if override.possible_graded_override is not None:
-                params['possible_graded'] = override.possible_graded_override
-        except PersistentSubsectionGradeOverride.DoesNotExist:
-            pass
-
         grade, _ = cls.objects.update_or_create(
             user_id=user_id,
             course_id=usage_key.course_key,
             usage_key=usage_key,
             defaults=params,
         )
         if first_attempted is not None and grade.first_attempted is None:
-            if waffle.waffle().is_enabled(waffle.ESTIMATE_FIRST_ATTEMPTED):
-                grade.first_attempted = first_attempted
-            else:
-                grade.first_attempted = now()
+            grade.first_attempted = first_attempted
             grade.save()
 
         cls._emit_grade_calculated_event(grade)
         return grade
 
     @classmethod
-    def _prepare_first_attempted_for_create(cls, params):
-        """
-        Update the value of 'first_attempted' to now() if we aren't
-        using score-based estimates.
-        """
-        if params['first_attempted'] is not None and not waffle.waffle().is_enabled(waffle.ESTIMATE_FIRST_ATTEMPTED):
-            params['first_attempted'] = now()
-
-    @classmethod
-    def create_grade(cls, **params):
-        """
-        Wrapper for objects.create.
-        """
-        cls._prepare_params_and_visible_blocks(params)
-        cls._prepare_first_attempted_for_create(params)
-
-        grade = cls.objects.create(**params)
-        cls._emit_grade_calculated_event(grade)
-        return grade
```

Comment:
**Note to reviewers:** `create_grade` is no longer needed by our latest usages - as we now just use `bulk_create_grades` and `update_or_create_grade`.